packages/esm-patient-notes-app/src/notes/visit-notes.resource.ts

```diff
@@ -104,10 +110,11 @@ export function useInfiniteVisits(patientUuid: string) {
   };
 }
 
-export function fetchConceptDiagnosisByName(searchTerm: string, diagnosisConceptClass: string) {
-  return openmrsObservableFetch<Array<Concept>>(
-    `${restBaseUrl}/concept?name=${searchTerm}&searchType=fuzzy&class=${diagnosisConceptClass}&name=&v=custom:(uuid,display)`,
-  ).pipe(map(({ data }) => data['results']));
+export function fetchDiagnosisConceptsByName(searchTerm: string, diagnosisConceptClass: string) {
+  const customRepresentation = 'custom:(uuid,display)';
+  const url = `${restBaseUrl}/concept?name=${searchTerm}&searchType=fuzzy&class=${diagnosisConceptClass}&v=${customRepresentation}`;
+
+  return openmrsFetch<Array<Concept>>(url).then(({ data }) => Promise.resolve(data['results']));
 }
```

Comment:
Can't we implement `useOpenMRSSWR` in this?

Reply:
So, refactoring the data fetching logic into a hook seems a lot less straightforward after looking at the ergonomics of how the primary diagnosis and secondary diagnosis search work.  It's easier when you only have one piece of state to run through the hook, but here we have both the primary and secondary diagnoses search, so at a minimum, we'd need to track more state for both. The code is fairly readable now, even without hooks, especially after switching to a promise-based implementation instead of using reactive logic via observables.

I'm happy to try and look at this in a separate PR though.